storage/backend/backends.go

```diff
@@ -40,20 +44,41 @@ func NewMountDriver(backend, mountpath string) (storage.MountDriver, error) {
 
 // NewCRUDDriver instantiates a CRUD Driver.
 func NewCRUDDriver(backend string) (storage.CRUDDriver, error) {
-	f, ok := CRUDDrivers[backend]
-	if !ok {
-		return nil, errored.Errorf("invalid CRUD driver backend: %q", backend)
-	}
+	f, _ := CRUDDrivers[backend]
 
 	return f()
 }
 
 // NewSnapshotDriver creates a SnapshotDriver based on the backend name.
 func NewSnapshotDriver(backend string) (storage.SnapshotDriver, error) {
-	f, ok := SnapshotDrivers[backend]
-	if !ok {
-		return nil, errored.Errorf("invalid snapshot driver backend: %q", backend)
-	}
+	f, _ := SnapshotDrivers[backend]
 
 	return f()
 }
+
+// NewDriver creates a driver based on driverType
+func NewDriver(backend string, driverType string, mountPath string, do storage.DriverOptions) error {
+	if backend != "" {
+		if driverType == CRUD {
+			if curd, err := NewCRUDDriver(backend); err != nil {
+				return err
+			} else if err := curd.Validate(do); err != nil {
+				return err
+			}
+		} else if driverType == Mount {
+			if mount, err := NewMountDriver(backend, mountPath); err != nil {
+				return err
+			} else if err := mount.Validate(do); err != nil {
+				return err
+			}
+		} else if driverType == Snapshot {
+			if snapshot, err := NewSnapshotDriver(backend); err != nil {
+				return err
+			} else if err := snapshot.Validate(do); err != nil {
+				return err
+			}
+		}
```

Comment:
I was thinking logic-wise... is it correct for there not to be an else case here?  Seems like we would want to raise an error if someone specifies a non-CRUD/-mount/-validate value in driverType rather than falling through to the final return.